### DISCUSSION TITLE
styles

### DISCUSSION THREAD

**Discussion by adrianpuiu** [seruva19/kubin#20] [General]
hi,
what file do i have to modify to add or remove styles?

**Comment by seruva19** [comment]
For now styles are hard-coded, but I will modify this extension so it would be possible to edit and save them in a file.

**Comment by adrianpuiu** [comment]
Do u have an ETA FOR this one ?

**Reply by seruva19** [reply]
I expect no later than Monday.

**Comment by adrianpuiu** [comment]
'none': '',
    'Greg': 'Greg Rutkowski, Alphonse Mucha, Beeple.',
    'bastien l. deharme painting': 'a painting in the style of bastien l. deharme.',
    'sam spratt': ' a painting in the style of sam spratt.',
    'flora borsi': 'a painting in the (style of flora borsi)..',
    'david downton': '(style of david downton).',
    'conrad rose': '(style of conrad roset).',
    'erica hopper': '(style of erica hopper).',
    'agnes cecile': '(style of agnes cecile).',
    'saturno butto': '(style of saturno butto).',
    'steampunk like': 'style of darek zabrocki and dan mcpharlin.',
    'paint dripping style basil gogos': 'paint dripping style of basil gogos.',
    'dan mcpharlin': 'style of dan mcpharlin.',
    'beautiful painting': '(style of augustus jansson).',
    'thomas saliot': '(style of thomas saliot).',
    'vintage retro poster': '(style of shepard fairey).',
    'android/cyborg portrait': 'in the style of android jones).',
    'adrianus eversen': 'a painting in the style of adrianus eversen).',
    'comic book style': ' in the style of sam kieth.',
    'armored portaits': 'in the style of chris cold.',
    'apocaliptic': 'in the style of scott listfield.',
    'painting by daniel f. gerhartz ': '(style of daniel f. gerhartz).',
    'edwin georgi': 'in the style of edwin georgi.',
    'russ mills': 'style of russ mills.',
    'heavy paint strokes by anna bocek': 'painting in the style of anna bocek, paint strokes and splatters..',
    'geometric painting by ryan hewett ': 'a painting in the style of ryan hewett)..',
    'futuristic city scapes': ' in the style of wadim kashin..',
    'vivid colors of felipe pantone)': ' in the style of felipe pantone..',
    'armor': ' in the style of allen williams..',
    'anna dittmann': 'style of anna dittmann..',
    'jesper ejsing': 'style of jesper ejsing..',
    'albert goodwin': 'impasto style of albert goodwin..',
    'paint splatter on backgound by erik jones': 'paint splatters on the backgound style of erik jones)..',
    'leonardo da vinci': 'a painting in the style of leonardo da vinci..',
    'peter howson': 'in the style of peter howson.',
    'comic book style 2': '(style of todd mcfarlane).',
    'vintage photo': 'a photo in the style of alex timmermans.',
    'alexandr averin': 'style of alexandr averin.',
    'anime': 'in anime style',
    'fear':'thick, impasto brush strokes and jarring contrasts of black against white. To achieve a similar effect digitally, noise or grain filters might also prove effective. For added impact, incorporating elements associated with horror - like skulls, spiders, or dripping blood - could enhance the overall sense of dread.',
    'anger': ' bold, sharp lines and deep shadows. Complementary warm colors, like reds and oranges, could heighten feelings of fury when included prominently in compositions. Digital techniques involving textures and gradients may also aid in evoking intense emotions',
    'joy':'expressed through bright hues, playful patterns, and fluid movements. Bold primary colors and organic forms could convey happiness effectively. Blurring edges softly, or utilizing painterly effects like splotches or blobs, may contribute to a carefree ambiance.',
    'love': 'illustrated through gentle curves, pastel tones, and delicate details. Romantic themes like hearts or flowers might naturally fit well within such imagery. Glitter or sparkling effects in digital media might further emphasize affectionate sentiments.',
    'hate': 'captured through sharp angles, harsh contrasts between lights and darks, and cold, desaturated palettes. Certain motifs associated with animosity, like knives or broken objects, could underscore detestation. Conversely, avoiding warm colors altogether might symbolize indifference or a lack of passion towards another person or topic',
    'detailed photo':	'4k, ultra HD, detailed phot',
    'cyberpunk': 'in cyberpunk style, futuristic cyberpunk',
    'kandinsky': 'painted by Vasily Kandinsky, abstractionis',
    'aivazovsky': 'painted by Aivazovsky',
    'malevich': 'Malevich, suprematism, avant-garde art, 20th century, geometric shapes , colorful, Russian avant-garde',
    'picasso': 'Cubist painting by Pablo Picasso, 1934, colorful',
    'goncharova': 'painted by Goncharova, Russian avant-garde, futurism, cubism, suprematism',
    'classicism':	'classicism painting, 17th century, trending on artstation, baroque painting',
    'renaissance': 'painting, renaissance old master royal collection, artstation',
    'oil painting':	'like oil painting',
    'pencil art':	'pencil art, pencil drawing, highly detailed',
    'digital painting':	'high quality, highly detailed, concept art, digital painting, by greg rutkowski trending on artstation',
    'medieval':	'medieval painting, 15th century, trending on artstation',
    'soviet cartoon':	'picture from soviet cartoons',
    '3d render': 'Unreal Engine rendering, 3d render, photorealistic, digital concept art, octane render, 4k HD',
    'cartoon': 'as cartoon, picture from cartoon',
    'studio photo':	'glamorous, emotional ,shot in the photo studio, professional studio lighting, backlit, rim lighting, 8k',
    'portrait photo':	'50mm portrait photography, hard rim lighting photography',
    'mosaic': 'as tile mosaic',
    'icon painting': 'in the style of a wooden christian medieval icon in the church',
    'khokhloma': 'in Russian style, Khokhloma, 16th century, marble, decorative, realistic',
    'new year':	'christmas, winter, x-mas, decorations, new year eve, snowflakes, 4k',

**Reply by seruva19** [reply]
That's cool. I might add it to default list perhaps? Or add function to download "upgraded" lists by url. No harm in having lots of styles :) 
I already allocated styles to external YAML file, now need to add UI for adding end editing styles, and then will publish the reworked extension. 
I also want to add field for storing and using negative prompts.
Here is another list I sometimes use in AUTOMATIC1111: https://docs.google.com/spreadsheets/d/1L6TI5_s2mVyHSMcph1oF4vgUjlFW3Um17zrd2GQs3Ec/edit?gid=10465324
(source: https://t.me/neuro_art0/1244)

**Reply by adrianpuiu** [reply]
nice , cant wait to put my hand on it to add more. ah , i know you have a lot to work on , but one more thing , can we have a small thumbnail for each style ?

**Reply by seruva19** [reply]
Yes, sure. For instance, I can make so that if there is a thumbnail with the same name as name of the style in the extension folder, it would automatically be fetched when selecting the style and shown in the UI.

**Comment by seruva19** [comment]
Latest commit to 'dev' branch modifies 'prompt-style' extension:

1. styles are read from external styles.yaml file (located in 'prompt-styles' extensions folder), which can be edited directly
2. there are also buttons for adding, editing and removing styles directly from UI (very basic, but it works xD).
3. now possible to add negative prompt to style

~~Not merging these changes into master yet, because latest commit also upgraded gradio version, and I want first to make sure nothing was broken :)~~

now merged into main https://github.com/seruva19/kubin/commit/8805591106e6bb2ad218c430e4ae6053d43c05e0

**Reply by adrianpuiu** [reply]
noice one . thanks a lot for the update! ill check tomorrow cause im to sleepy atm

**Reply by adrianpuiu** [reply]
I've just had a look over the code and frankly i'm impressed by the fact you also included negative prompts in the styles. also happy you've put the image settings into the accordion.  Well done.
PS : don't forget about adding the send to upscale button in the image gallery ( from t2i, i2i, inpainting -> upscale ) if possible ..

**Reply by seruva19** [reply]
There must be 'Send to Upscaler' button in all tabs, don't you have it?

**Comment by adrianpuiu** [comment]
I only see with the left eye :)) .. with one I look and with the other one
I convince my self.. yes there is . sorry
